### PR TITLE
BZ-1979431: Clarifying behavior of the default node selector

### PR DESCRIPTION
**Pull request by bergerhoffer**
https://bugzilla.redhat.com/show_bug.cgi?id=1979431

Preview: https://deploy-preview-34928--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/cluster-tasks.html#creating-an-infra-node_post-install-cluster-tasks

### CODE REVIEW

**Comment by ingvagabund** [comment]
lgtm

**Comment by bergerhoffer** [comment]
@kasturinarra Can you please review this update? Related to https://bugzilla.redhat.com/show_bug.cgi?id=1979431.

Preview: https://deploy-preview-34928--osdocs.netlify.app/openshift-enterprise/latest/post_installation_configuration/cluster-tasks.html#creating-an-infra-node_post-install-cluster-tasks

File: modules/creating-an-infra-node.adoc [hunk 1]
```diff
@@ -35,11 +35,37 @@ $ oc label node <node-name> node-role.kubernetes.io/infra=""
...
 ----
 
-. Create a default node selector so that pods without a node selector are assigned a subset of nodes to be deployed on, for example by default deployment in worker nodes. As an example, the `defaultNodeSelector` to deploy pods on worker nodes by default would look like:
+. Create a default cluster-wide node selector. The default node selector is applied to pods created in all namespaces. This creates an intersection with any existing node selectors on a pod, which additionally constrains the pod's selector.
++
+[IMPORTANT]
+====
+If the default node selector key conflicts with the key of a pod's label, then the default node selector is not applied.
+
+However, be careful not to set a default node selector that will cause a situation where a pod becomes unschedulable. For example, setting the default node selector to a specific node role, such as `node-role.kubernetes.io/infra=""`, when a pod's label is set to a different node role, such as `node-role.kubernetes.io/master=""`, can cause the pod to become unschedulable. For this reason, it is not recommended to set the default node selector to specific node roles.
```
Comment (bobfuru) [inline modules/creating-an-infra-node.adoc:44]: Not sure how you feel about shortening the first sentence, but perhaps it is more direct to say something like, "However, do not to set a default node selector that causes a pod to become unschedulable" (or "that can cause"). 🤷

**Review by bobfuru — COMMENTED** [review]
Overall, LGTM! Left a couple of comments for consideration, one of which is to amend content not actually part of this PR, if you're so inclined. 😀

File: modules/creating-an-infra-node.adoc [hunk 2]
```diff
@@ -35,11 +35,37 @@ $ oc label node <node-name> node-role.kubernetes.io/infra=""
 $ oc get nodes
 ----
 
-. Create a default node selector so that pods without a node selector are assigned a subset of nodes to be deployed on, for example by default deployment in worker nodes. As an example, the `defaultNodeSelector` to deploy pods on worker nodes by default would look like:
+. Create a default cluster-wide node selector. The default node selector is applied to pods created in all namespaces. This creates an intersection with any existing node selectors on a pod, which additionally constrains the pod's selector.
```
[the remark this answers is not in the document]
Comment (bergerhoffer) [reply modules/creating-an-infra-node.adoc:38]: Thanks! I'll take care of the control plane fix, but I'm going to leave the doc ref alone. I'm pretty against doing these kinds of implicit dependencies like this without a link. Definitely don't want my name attached to even an update to it :wink:

**Pull request merged**